src/config/providedAssets.ts

```diff
@@ -0,0 +1,19 @@
+import UserConfig from '.';
+
+export default () => {
+    const userConfig = new UserConfig().getUserConfig();
+
+    const pairs = userConfig.PAIRS;
+
+    return Object.keys(pairs).reduce((result, pair) => {
+        const asset = pair.split('-')[1];
+
+        if (asset) {
+            if (!result[asset]) {
+                result[asset] = true;
```

Comment:
maybe you need            
 `if (!result[quote]) {
                result[quote] = true;
 }`

someone can have pair ETH-BTC, but BTC-ETH might be missing

Reply:
That's not the case. When a LP chooses to support ETH-BTC , this means that the LP will provide BTC in exchange for ETH. Which means that the LP needs to report only his BTC balance and not ETH balance. And that's where the provided assets are used.